### PR TITLE
multi-gpu and training/evaluation speed up

### PR DESCRIPTION
**Pull request by ekmb**

### CODE REVIEW

File: examples/nlp/token_classification/punctuation_capitalization.py [hunk 1]
```diff
@@ -164,11 +190,11 @@ def create_pipeline(
     capit_label_ids=None,
     ignore_extra_tokens=args.ignore_extra_tokens,
     ignore_start_end=args.ignore_start_end,
-    use_cache=args.use_cache,
+    overwrite_processed_files=args.overwrite_processed_files,
```
Comment (VahidooX) [inline examples/nlp/token_classification/punctuation_capitalization.py:193]: Would you please print something when overwrite_processed_files is false and the processed files are already exist, to let the user know that you have not created the files?

File: examples/nlp/token_classification/punctuation_capitalization.py [hunk 2]
```diff
@@ -110,14 +128,21 @@
...
+    "--loss_log_freq", default=50, type=int, help="Frequency of logging loss values, '-1' - at the end of the epoch",
+)
 parser.add_argument(
     "--use_weighted_loss_punct",
     action='store_true',
     help="Flag to indicate whether to use weighted loss \
                     to mitigate classs unbalancing for the punctuation task",
 )
-
+parser.add_argument("--project", default=None, type=str, help='Project name for tracking with Weights and Biases')
```
Comment (VahidooX) [inline examples/nlp/token_classification/punctuation_capitalization.py:143]: How about project_name or something to show it is for wandb? like wandb_project and wandb_exp? or wb_project_name and wb_exp_name?

Comment (ekmb) [reply examples/nlp/token_classification/punctuation_capitalization.py]: done

File: examples/nlp/token_classification/punctuation_capitalization.py (hunk 1 above)
Comment (ekmb) [reply examples/nlp/token_classification/punctuation_capitalization.py:207]: Such info message is logged from the dataset

File: nemo/collections/nlp/nm/data_layers/punctuation_capitalization_datalayer.py [hunk 3]
```diff
@@ -115,6 +115,6 @@ def __init__(
             'capit_label_ids': capit_label_ids,
             'ignore_extra_tokens': ignore_extra_tokens,
             'ignore_start_end': ignore_start_end,
-            'use_cache': use_cache,
+            'overwrite_processed_files': overwrite_processed_files,
         }
-        super().__init__(dataset_type, dataset_params, batch_size, shuffle=shuffle)
+        super().__init__(dataset_type, dataset_params, batch_size, shuffle=shuffle, num_workers=2, pin_memory=True)
```
[the remark this answers is not in the document]
Comment (ekmb) [reply nemo/collections/nlp/nm/data_layers/punctuation_capitalization_datalayer.py]: done

**Pull request merged**